Api/Controllers/EventsController.cs

```diff
@@ -184,5 +184,23 @@ public async Task<ActionResult> DeleteEvent(int eventId)
 
             return OkOrErrors(await service.DeleteEventAsync(eventId, user));
         }
+
+        /// <summary>
+        /// Get all events that fulfill the requirements
+        /// </summary>
+        /// <param name="request">dto with requirements</param>
+        /// <returns>list of events</returns>
+        [HttpGet]
+        [ProducesResponseType(200), ProducesResponseType(401)]
+        public async Task<ActionResult<List<EventVM>>> GetEvents(GetEventsRequest request)
```

Comment:
```suggestion
        public async Task<ActionResult<List<EventVM>>> GetEvents([FromQuery] GetEventsRequest request)
```
inaczej próbuje odczytać z body